server/auth/types/openid/openid_auth.ts

```diff
@@ -128,11 +127,14 @@ export class OpenIdAuthentication extends AuthenticationType {
   }
 
   private generateNextUrl(request: OpenSearchDashboardsRequest): string {
-    const path = getRedirectUrl({
+    let path = getRedirectUrl({
       request,
       basePath: this.coreSetup.http.basePath.serverBasePath,
       nextUrl: request.url.pathname || '/app/opensearch-dashboards',
     });
+    if (request.url.search) {
+      path += request.url.search;
```

Comment:
Thank you @KleinSebastian. I started the CI checks again. This PR looks good to me and has extensive tests. Thank you for adding those?

Quick question: https://github.com/opensearch-project/security-dashboards-plugin/issues/2119 also reports that the `#hash` part of a url is getting dropped. I see that this fix will carry url params over, but will it also carry the hash if there is one?

Reply:
Carrying over all the data after the hash was already implemented in a previous PR, but was bugged . With the small change in https://github.com/opensearch-project/security-dashboards-plugin/pull/2140/files/d5bb3126b3610f3264b112d4b8adb12a8f18dcd7#diff-55f93c47c667faee3f68b8b8f5e98157fb36eefdc7b856355d5a9b4350526070R104 the functionality of capturing and restoring the hash part is working as expected.

Reply:
Got it. Thank you for confirming!